### PR TITLE
Add Function Flag Properties

### PR DESCRIPTION
**Pull request by cledoux**
Fix for #64 

* Adds one property to `function.Function` for each function flag documented at https://www.hex-rays.com/products/ida/support/sdkdoc/group___f_u_n_c__.html
* Uses the dictionary `function.FUNC_FLAGS`  to provide a list of function flags directly in the code.

### CODE REVIEW

**Comment by cledoux** [comment]
Like in my last PR, I'm not sure how to provide test cases for this.  I loaded a function in IDA and ensured that the new properties didn't throw any errors, but I don't have anything automated to provide.

If you can provide guidelines as to how to test, I'd be happy to do ammend the PR.

**Comment by cledoux** [comment]
Adding a new commit instead of rebasing to highlight the change.  I can rebase and squash commits once you are happy with all the changes.

File: sark/code/function.py [hunk 1]
```diff
@@ -50,6 +80,81 @@ def __repr__(self):
             repeat=repr(self.repeat))
 
 
+class FunctionFlags(long):
```
Comment (tmr232) [inline sark/code/function.py]: I don't like inheriting from `long`.
That said, I'm not sure how to handle the current situation without breaking old code. Maybe a new name for this member?

Comment (cledoux) [reply sark/code/function.py]: I think a new name is the only other way to maintain backwards compatibility. I can't think of a good name for it, though. The only thing I can think of is `flag_checks` and that seems overly verbose.  Any suggestions?

Comment (cledoux) [reply sark/code/function.py]: Also, If you don't mind me asking, why don't you like inheriting from `long`?  I'm not trying to disagree with you, just trying to learn.

**Comment by cledoux** [comment]
@tmr232  What do you think of this approach?  Using a mixin to provide all the `is_*` functions without cluttering the class itself?

If you like this approach, I can rebase into a single commit with just these changes.

**Comment by tmr232** [comment]
This looks good. Sorry for taking so long.

**Pull request merged**